sql/src/main/java/io/crate/operation/reference/sys/node/NodeHostnameExpression.java

```diff
@@ -22,23 +22,26 @@
 package io.crate.operation.reference.sys.node;
 
 import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.cluster.ClusterService;
 import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.node.service.NodeService;
 
 public class NodeHostnameExpression extends SysNodeExpression<BytesRef> {
 
     public static final String NAME = "hostname";
 
-    private final NodeService nodeService;
+    private final ClusterService clusterService;
 
     @Inject
-    public NodeHostnameExpression(NodeService nodeService) {
-        this.nodeService = nodeService;
+    public NodeHostnameExpression(ClusterService clusterService) {
+        this.clusterService = clusterService;
     }
 
     @Override
     public BytesRef value() {
-        return new BytesRef(nodeService.stats().getHostname());
+        if (clusterService.localNode() != null) {
```

Comment:
store to local variable and then check for null, instead of getting it twice from the cluster service